src/PolicyEngine.jsx

```diff
@@ -239,7 +245,7 @@ export default function PolicyEngine() {
     if (countryId && isAuthenticated && user?.sub) {
       fetchUserProfile().then((userProfile) => setUserProfile(userProfile));
     }
-  }, [countryId, user?.sub, isAuthenticated]);
+  }, [countryId, user?.sub, isAuthenticated, authenticatedApiCall]);
```

Comment:
__question, non-blocking:__ Theoretically, at a future step, would we expect to extend `authenticatedApiCall` to all calls, then be required to add the function to all effect hook dep arrays?

Or perhaps another way to frame this: we clearly have too many effect hook calls

Reply:
Yes, my plan is to switch everything over to auto-authenticated API calls and that will result in everything making an API call using a hook.

I haven't taken a deep dive into the react and use of hooks yet, but I agree there is too much in general going on in our react components.